src/Illuminate/Database/Query/Builder.php

```diff
@@ -692,6 +692,10 @@ public function where($column, $operator = null, $value = null, $boolean = 'and'
         // and can add them each as a where clause. We will maintain the boolean we
         // received when the method was called and pass it into the nested where.
         if (is_array($column)) {
+            if (! is_null($operator)) {
+                throw new \RuntimeException("When the operator is not null, the column shouldn't be array");
+            }
```

Comment:
Maybe I am missing something but I don't really understand what this prevents? When the `$column` parameter is an array it doesn't matter what the `$operator` parameter is set to. It's not used or passed to `addArrayOfWheres`.

Reply:
If the column is array, we always don't write operator, like the code below
```
->where($wheres);
```

Reply:
But it doesn't matter if you set the `$operator` argument or not. The "issue" is the same, you will still be able to overwrite the operator in the array. Thats because the fourth item in the array specifies the operator. A fix could be that the fourth argument is validated to be a valid operator.